Python/Product/PythonTools/PythonTools/WorkspaceInfoBarManager.cs

```diff
@@ -50,17 +52,22 @@ class WorkspaceInfoBarManager : IVsRunningDocTableEvents, IDisposable {
 
         private void OnWorkspaceInitialized(object sender, PythonWorkspaceContextEventArgs e) {
             var workspace = e.Workspace;
+            _infoBarCheckTriggered = false;
 
             _packageInstallInfoBar = new PackageInstallWorkspaceInfoBar(_serviceProvider, workspace);
             _condaEnvCreateInfoBar = new CondaEnvCreateWorkspaceInfoBar(_serviceProvider, workspace);
             _virtualEnvCreateInfoBar = new VirtualEnvCreateWorkspaceInfoBar(_serviceProvider, workspace);
             _testFrameworkInfoBar = new TestFrameworkWorkspaceInfoBar(_serviceProvider, workspace);
+            _pythonVersionNotSupportedInfoBar = new PythonNotSupportedInfoBar(_serviceProvider, InfoBarContexts.Workspace, () => workspace.CurrentFactory);
 
             workspace.AddActionOnClose(_packageInstallInfoBar, (obj => ((PythonInfoBar)obj).Dispose()));
             workspace.AddActionOnClose(_condaEnvCreateInfoBar, (obj => ((PythonInfoBar)obj).Dispose()));
             workspace.AddActionOnClose(_virtualEnvCreateInfoBar, (obj => ((PythonInfoBar)obj).Dispose()));
             workspace.AddActionOnClose(_testFrameworkInfoBar, (obj => ((PythonInfoBar)obj).Dispose()));
-            _infoBarCheckTriggered = false;
+            workspace.AddActionOnClose(_pythonVersionNotSupportedInfoBar, (obj => ((PythonInfoBar)obj).Dispose()));
+
+            workspace.ActiveInterpreterChanged += TriggerPythonNotSupportedInforBar;
+            workspace.AddActionOnClose(_pythonVersionNotSupportedInfoBar, (obj) => workspace.ActiveInterpreterChanged -= TriggerPythonNotSupportedInforBar);
```

Comment:
You should check the implementation, but I think the first param is a key, and it will overwrite the previous `AddActionOnClose`, which would mean dispose wouldn't get called, so should be easy to verify.